src/orange/OrangeData.hh

```diff
@@ -273,6 +313,7 @@ struct OrangeParamsData
     Items<SurfaceType> surface_types;
     Items<Connectivity> connectivities;
     Items<VolumeRecord> volume_records;
+    Items<BIHNode> bih_nodes;
```

Comment:
I think you should have a separate header file and class for `BIHData`, just like we do for `TwodGridData`, because it could conceivably be used for other applications other than a full-fledged ORANGE geometry. You can use composition to just have a 
```c++
BIHNodeData<W, M> bih;
```
member in OrangeParams.

Reply:
Both `TwodGridData` and `BIHNodeData` (which I think I am going to call `BIHParams`) only store `ItemRanges` so I don't think they need to be templated on `W, M`, as they don't store actual `Collections`.